server/src/test/java/org/elasticsearch/cluster/metadata/IndexAbstractionTests.java

```diff
@@ -116,13 +118,82 @@ public void testHiddenAliasValidation() {
         }
     }
 
-    private IndexMetadata buildIndexWithAlias(String indexName, String aliasName, @Nullable Boolean aliasIsHidden) {
+    public void testSystemAliasValidation() {
```

Comment:
I'm not generally a fan of long tests that check lots of cases. Can we usefully split up this method into a series of more targeted test cases, ideally with a Javadoc comment that explains what is the test is for ([see point 9 here](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#the-long-version)).

Reply:
I've split this test and the one in `IndexNameExpressionResolverTests` into multiple tests. I have not added Javadoc to the split test cases though, as the test cases ended up fairly small and I was finding in attempting to write Javadoc for them that the Javadoc largely was just repeating the test names.